build_pros.cmake

```diff
@@ -26,44 +26,16 @@ endif()
 
 # ======== Optimizations and warnings ========
 
-# Strip garbage
-if(APPLE)
-    string(REGEX REPLACE "-O3" ""
-        CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
-    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O2")
-    set(LINK_FLAGS_RELEASE  "${LINK_FLAGS_RELEASE} -dead_strip")
-elseif(NOT MSVC)
-    if(EMSCRIPTEN)
-        set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -Os -fdata-sections -ffunction-sections")
-        if("${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
-            set(LINK_FLAGS_RELEASE  "${LINK_FLAGS_RELEASE} -dead_strip")
-        else()
-            set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -Wl,--gc-sections -Wl,-s")
-            set(LINK_FLAGS_RELEASE  "${LINK_FLAGS_RELEASE} -Wl,--gc-sections -Wl,-s")
-        endif()
-    else()
-        string(REGEX REPLACE "-O3" ""
-            CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
-        set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O2 -fdata-sections -ffunction-sections")
-        if(ANDROID)
-            set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -funwind-tables")
-        elseif(NOT "${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
-            set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -s -Wl,--gc-sections -Wl,-s")
-            set(LINK_FLAGS_RELEASE  "${LINK_FLAGS_RELEASE} -Wl,--gc-sections -Wl,-s")
-        else()
-            set(LINK_FLAGS_RELEASE  "${LINK_FLAGS_RELEASE} -dead_strip")
-        endif()
-    endif()
-endif()
-
 # Global optimization flags
 if(NOT MSVC)
     set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -fno-omit-frame-pointer")
+    set(CMAKE_C_FLAGS_MINSIZEREL "${CMAKE_C_FLAGS_MINSIZEREL} -fno-omit-frame-pointer")
```

Comment:
Is there a specific reason for not omitting the fame pointer?

Reply:
This allows making stack traces that will show something useful. Otherwise, stack traces will be useless.

Reply:
How meaningful are stack traces in optimized builds, anyway?

Reply:
Mainly to allow users of applications to generate reports and send them to developers as a hint where the failure was happen

Reply:
I'd say that it's blah and vote for removal.. @AliceLR ?